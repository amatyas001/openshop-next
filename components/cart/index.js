export * from './CartContent/CartContent';
export * from './CartAmount/CartAmount';
export * from './CartButton/CartButton';
export * from './CartItem/CartItem';
export * from './CartPanel/CartPanel';
