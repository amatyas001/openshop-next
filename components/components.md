Elements are composed by [Chakra-UI](https://chakra-ui.com) components and preserve the library features so [Style Props](https://chakra-ui.com/style-props) can be passed to them and accessibility remains optimized.

> _Style Props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components._

For more information visit the [Chakra UI](https://chakra-ui.com) documentation.

Components are divided into two main category: _wrappers_ and _single responsibility_ elements. Wrapper components does not contain business logic, however there are special cases like _handling lifted states to keep sensitive data locally_ and not to populate them to the Redux store which is persisted on the machine.

**State Dependencies** and **Wrapped Components** are documented in each description to give brief information about the context of posible interactions of the actual component. _Examples_ prepared with proper state and may render extra controls to test different cases with ease.
