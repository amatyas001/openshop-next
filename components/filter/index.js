export * from './FilterButton/FilterButton';
export * from './FilterColor/FilterColor';
export * from './FilterPrice/FilterPrice';
export * from './FilterName/FilterName';
export * from './FilterPanel/FilterPanel';
export * from './FilterReset/FilterReset';
