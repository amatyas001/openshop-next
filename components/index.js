export * from './cart';
export * from './filter';
export * from './product';
export * from './layout';
export * from './payment';
export * from './shared';
