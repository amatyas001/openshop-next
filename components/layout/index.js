export * from './Content/Content';
export * from './Footer/Footer';
export * from './ProductsTopBar/ProductsTopBar';
export * from './Navbar/Navbar';
export * from './ScrollTop/ScrollTop';
