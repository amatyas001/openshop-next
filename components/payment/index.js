export * from './PaymentCancelled/PaymentCancelled';
export * from './PaymentConfirm/PaymentConfirm';
export * from './PaymentConfirmCard/PaymentConfirmCard';
export * from './PaymentConfirmControls/PaymentConfirmControls';
export * from './PaymentConfirmDetails/PaymentConfirmDetails';
export * from './PaymentError/PaymentError';
export * from './PaymentForm/PaymentForm';
export * from './PaymentFormControls/PaymentFormControls';
export * from './PaymentProgress/PaymentProgress';
export * from './PaymentReview/PaymentReview';
export * from './PaymentSuccess/PaymentSuccess';
