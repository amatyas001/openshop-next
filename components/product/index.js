export * from './ProductCard/ProductCard';
export * from './ProductControls/ProductControls';
export * from './ProductControlsColor/ProductControlsColor';
export * from './ProductControlsAmount/ProductControlsAmount';
export * from './ProductControlsButton/ProductControlsButton';
export * from './ProductControlsSize/ProductControlsSize';
