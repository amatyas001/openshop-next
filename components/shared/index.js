export * from './Button/Button';
export * from './ButtonIcon/ButtonIcon';
export * from './Field/Field';
export * from './Image/Image';
export * from './Panel/Panel';
export * from './Spinner/Spinner';
