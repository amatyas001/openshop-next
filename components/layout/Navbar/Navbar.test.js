import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import { Navbar } from '@app/components';
import { INITIAL_STATE } from '@app/config';

const mockStore = createStore([]);

describe('<Navbar />', () => {
  let tree, store;

  beforeAll(() => {
    store = mockStore(INITIAL_STATE);

    act(() => {
      tree = create(
        <Provider store={store}>
          <Navbar />
        </Provider>
      );
    });
  });

  it('should render without props', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
