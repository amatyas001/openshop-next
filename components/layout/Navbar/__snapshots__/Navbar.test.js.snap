// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render without props 1`] = `
<header
  className="css-2d7au8"
  data-cy="nav-container"
>
  <a
    className="css-1444nv"
    data-testid="navbar-brand"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    openshop
  </a>
  <nav
    aria-label="Main"
    className="css-ntite"
    data-cy="nav-navbar"
    id="header-nav"
  >
    <ul
      style={
        Object {
          "margin": 0,
          "padding": 0,
        }
      }
    >
      <li
        data-cy="nav-item"
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
          }
        }
      >
        <button
          className="css-1gzeiec"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          tabIndex="0"
          type="button"
        >
          home
        </button>
      </li>
      <li
        data-cy="nav-item"
        style={
          Object {
            "display": "inline-block",
            "listStyleType": "none",
          }
        }
      >
        <button
          className="css-1gzeiec"
          href="/products"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          tabIndex="0"
          type="button"
        >
          products
        </button>
      </li>
    </ul>
  </nav>
  <section
    className="css-j7qwjs"
    id="cart"
    role="complementary"
  >
    <button
      aria-label="cart-toggler"
      className="css-1vus3us"
      data-testid="cart-panel-button"
      href="#void"
      id="cart-toggler"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex="0"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "marginBottom": "-5px",
            "marginRight": "5px",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
      0 item(s)
    </button>
    <div
      aria-hidden={true}
      className="rah-static rah-static--height-zero css-vppuuz"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "WebkitTransition": "opacity 250ms ease 0ms",
            "opacity": 0,
            "transition": "opacity 250ms ease 0ms",
          }
        }
      >
        <div
          className="css-1hpslha"
        >
          <div
            className="css-ugjhak"
          >
            <button
              className="css-1ks4ucs"
              data-testid="cart-panel-button-close"
              href="#void"
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex="0"
              type="button"
            >
              close
            </button>
            <button
              className="css-48tyil"
              data-testid="cart-panel-button-continue"
              disabled={true}
              href="/checkout"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              tabIndex="0"
              type="button"
            >
              continue
            </button>
            <div
              className="css-ufscxh"
              data-testid="cart-panel-amount"
            >
              <p
                className="css-dptndp"
              >
                TOTAL
              </p>
              <strong
                className="css-lk14dv"
                data-testid="cart-amount-value"
              >
                0.00
                Â $
              </strong>
            </div>
          </div>
          <ul
            className="css-1q2qbgw"
            data-testid="cart-panel-content"
            role="list"
          />
        </div>
      </div>
    </div>
  </section>
</header>
`;
